packages/renderer/src/lib/preferences/PreferencesResourcesRendering.svelte

```diff
@@ -275,51 +298,116 @@ function isContainerConnectionStatusInProgress(
                       <Tooltip tip="Start" bottom>
                         <button
                           aria-label="Start"
-                          class="{container.status !== 'stopped' ||
-                          isContainerConnectionStatusInProgress(provider, container)
+                          class="ml-5 mr-2.5 my-2 {isContainerConnectionActionInProgress(
```

Comment:
Maybe worth having this is a separate method/function

Reply:
You mean generating the whole classes list? Or just
```
 {isContainerConnectionActionInProgress(
                            containerConnectionStatus,
                            provider,
                            container,
                          ) || container.status !== 'stopped'
                            ? 'text-gray-700 cursor-not-allowed'
                            : 'hover:text-gray-400'}
```

Reply:
I think anyway there is a code smell to use functions in `classes` as it may not be called when we think

Reply:
code smell:
--> it may lead to too many calls. Usually attributes should only be refreshed when there is a change, not call functions all the time.
--> function may not be called

Reply:
I don't get this and i'm unsure how to do it differently so any clarification is appreciated
 
1) my understanding is that the function will be called when there is an update of its reactive params. The containerConnectionStatus gets only updated in two situations (when a button is clicked and when the status of a container is updated). In this case all buttons have to be re-verified as i cannot know what container has been updated. 
2) why may not be called?

Reply:
What I'm saying is that the preferable way is to have a structure containing all the state and display the attributes from the state.


Like:

You have an array of objects and you  have a field `isInProgress`. You don't call isInProgress method, you just use the field.

Reply:
> You mean generating the whole classes list? Or just
> 
> ```
>  {isContainerConnectionActionInProgress(
>                             containerConnectionStatus,
>                             provider,
>                             container,
>                           ) || container.status !== 'stopped'
>                             ? 'text-gray-700 cursor-not-allowed'
>                             : 'hover:text-gray-400'}
> ```

Not the method that return a single class name for a specified state